tests/hw7.py

```diff
@@ -0,0 +1,82 @@
+        wb = self.browser
+        expected = "Username cannot be empty"
+        wb.find_element(By.ID, "btnLogin").click()
+        actual = wb.find_element(By.XPATH, "//div/span[@id='spanMessage']").text
+        self.assertEqual(expected, actual)
+
+    def test_valid_username_and_invalid_password(self):
+        wb = self.browser
+        expected = "Password cannot be empty"
+        wb.find_element(By.ID, "txtUsername").send_keys(ADMIN_USER)
+        wb.find_element(By.ID, "btnLogin").click()
+        actual = wb.find_element(By.XPATH, "//div/span[@id='spanMessage']").text
+        self.assertEqual(expected, actual)
+
+    def test_invalid_username_and_valid_password(self):
+        wb = self.browser
+        expected = "Username cannot be empty"
+        wb.find_element(By.ID, "txtPassword").send_keys(DEFAULT_PASSWORD)
+        wb.find_element(By.ID, "btnLogin").click()
+        actual = wb.find_element(By.XPATH, "//div/span[@id='spanMessage']").text
+        self.assertEqual(expected, actual)
+
+    # Part 2: Test for sorting table values by First (Middle) Name column
+    def test_sorting_table_values(self):
+        wb = self.browser
+        authentification(wb)
+        WebDriverWait(self.browser, 5).until(expected_conditions.url_contains("/pim/viewEmployeeList"))
+        wb.find_element(By.XPATH, "//thead/tr/th[3]/a").click()
+        WebDriverWait(self.browser, 5).until(expected_conditions.url_contains(
+            "viewEmployeeList?sortField=firstMiddleName&sortOrder=ASC"))
+        actual_list_of_web_elements = wb.find_elements(By.XPATH, "//tbody/tr/td[3]/a")
+        actual_list = []
+        for x in actual_list_of_web_elements:
+            actual_list.append(x.text)
+        sorted_list = sorted(actual_list)
+        self.assertEqual(sorted_list, actual_list)
+
+    # Part 3: Bonus challenge
+    # Sorting table values by First (Middle) Name column
+    # for all pages of employee records
+    def test_table_value_for_all_pages(self):
+        wb = self.browser
+        authentification(wb)
+        WebDriverWait(self.browser, 5).until(expected_conditions.url_contains("/pim/viewEmployeeList"))
+        wb.find_element(By.XPATH, "//thead/tr/th[3]/a").click()
+        WebDriverWait(self.browser, 5).until(expected_conditions.url_contains(
+            "viewEmployeeList?sortField=firstMiddleName&sortOrder=ASC"))
+        page_count = int(wb.find_element(By.XPATH, "//div[@class='top']/ul/li[1]").text.split()[-1])
+        for x in range(int(page_count / 50) + 1):
+            if x > 0:
+                wb.find_element(By.XPATH, "//div[@class='top']/ul/li[6]/a").click()
+            actual_list_of_web_elements = wb.find_elements(By.XPATH, "//tbody/tr/td[3]/a")
+            actual_list = []
+            for value in actual_list_of_web_elements:
+                actual_list.append(value.text)
+            sorted_list = sorted(actual_list)
+            self.assertEqual(sorted_list, actual_list)
```

Comment:
Good solution, however it can become problematic if there are dozens of pages.

Also consider the outcome if there is only one page of results, in which case pagination is not displayed at all. 😉